langserver/server.go

```diff
@@ -20,20 +20,24 @@ package langserver
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"net"
 	"net/http"
 	"os"
 	"strings"
 	"sync"
 
+	"github.com/blang/semver"
 	"github.com/pkg/errors"
 
 	"github.com/prometheus-community/promql-langserver/internal/vendored/go-tools/jsonrpc2"
 	"github.com/prometheus-community/promql-langserver/internal/vendored/go-tools/lsp/protocol"
 	"github.com/prometheus-community/promql-langserver/langserver/cache"
 	"github.com/prometheus/client_golang/api"
 	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	api_v1 "github.com/prometheus/prometheus/web/api/v1"
```

Comment:
You shouldn't include this package.

It pulls in large parts of Prometheus as a dependency and makes the binary four times as large.

Also the crossbuild now runs out of memory and gets killed.

Reply:
Just copy the struct definitions you need and mention why you did that in a comment.